lib/internal/Magento/Framework/HTTP/Adapter/Curl.php

```diff
@@ -175,6 +175,9 @@ public function write($method, $url, $http_ver = '1.1', $headers = [], $body = '
         if ($method == \Zend_Http_Client::POST) {
             curl_setopt($this->_getResource(), CURLOPT_POST, true);
             curl_setopt($this->_getResource(), CURLOPT_POSTFIELDS, $body);
+        } elseif ($method == \Zend_Http_Client::PUT) {
+            curl_setopt($this->_getResource(), CURLOPT_CUSTOMREQUEST, 'PUT');
```

Comment:
There's a `CURLOPT_PUT` constant and with PUT requests you should be using `CURLOPT_INFILE` and `CURLOPT_INFILESIZE`.